akka-http-core/src/main/scala/akka/http/impl/engine/parsing/BodyPartParser.scala

```diff
@@ -144,14 +131,15 @@ private[http] final class BodyPartParser(
       def parsePreamble(input: ByteString, offset: Int): StateResult =
         try {
           @tailrec def rec(index: Int): StateResult = {
-            val needleEnd = boyerMoore.nextIndex(input, index) + needle.length
-            if (crlf(input, needleEnd)) parseHeaderLines(input, needleEnd + 2)
+            val needleEnd = eolConfiguration.boyerMoore.nextIndex(input, index) + eolConfiguration.needle.length
+            if (eolConfiguration.isEndOfLine(input, needleEnd)) parseHeaderLines(input, needleEnd + eolConfiguration.eolLength)
             else if (doubleDash(input, needleEnd)) setShouldTerminate()
             else rec(needleEnd)
           }
+          eolConfiguration = EndOfLineConfiguration(input, boundary)
```

Comment:
This doesn't match `var eolConfiguration `'s comment

```
Will be override in tryParseInitialBoundary, but defined here as norm version to avoid NPE
```